src/Stratis.Bitcoin.Features.SmartContracts/PoW/SmartContractBlockDefinition.cs

```diff
@@ -83,7 +86,9 @@ public override void AddToBlock(TxMempoolEntry mempoolEntry)
             {
                 this.logger.LogTrace("Transaction contains smart contract information.");
 
-                // We HAVE to first execute the smart contract contained in the transaction
+                if (this.blockGasConsumed >= GasPerBlockLimit) return;
```

Comment:
This is the 1 part I'm cautious of, I think this is right, but just want to make sure in tests that it doesn't think the tx was included in one block or the other when it wasn't

Reply:
@codingupastorm updated PR. Added integration test.